libraries/AP_NavEKF2/AP_NavEKF2.h

```diff
@@ -493,6 +493,28 @@ class NavEKF2 {
     // time of last lane switch
     uint32_t lastLaneSwitch_ms;
 
+    enum class INIT_FAILURES {
+        OK,
+        NO_MEM, 
+        NO_ENABLE, 
+        NO_IMUS, 
+        NO_MASK, 
+        NO_ALLOC,
+        UNKNOWN,
+        NUM_INIT_FAILURES
+    };
+    // initialization failure reasons
+    const char* init_failure_reason[int(INIT_FAILURES::NUM_INIT_FAILURES)] {
+        "OK",
```

Comment:
I think we may need to prepend "EKF2:" or something similar or the pre-arm failure will be, "PreArm: OK" or "PreArm:memory allocation failed".

Maybe the first option should be "Unknown"?

Reply:
done